src/AstProfileUse.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2018, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstProfileUse.cpp
+ *
+ * Implements an AstAnalysis that provides profile information
+ * from a profile log file for profile-guided optimisations.
+ *
+ ***********************************************************************/
+
+#include "AstProfileUse.h"
+#include "AstClause.h"
+#include "AstLiteral.h"
+#include "AstProgram.h"
+#include "AstRelation.h"
+#include "AstRelationIdentifier.h"
+#include "AstTranslationUnit.h"
+#include "AstUtils.h"
+#include "AstVisitor.h"
+#include "Global.h"
+#include "Util.h"
+#include <algorithm>
+#include <iterator>
+#include <set>
+#include <string>
+
+namespace souffle {
+
+AstProfileUse::AstProfileUse() : programRun(std::make_shared<profile::ProgramRun>(profile::ProgramRun())) {}
+
+void AstProfileUse::run(const AstTranslationUnit& translationUnit) {
+    if (Global::config().has("profile-use")) {
+        std::string filename = Global::config().get("profile-use");
+        profile::Reader(filename, programRun).processFile();
```

Comment:
Assuming this grabs the profile output from the `--profile=blah.out` option, which is great because it has a lot of useful information.

Can there be an option for a more reduced format so that, without actual profile information, the user can give in estimated sizes based on domain-specific knowledge, etc.? E.g. a tab-delimited file like:

```
somerelation 1000
someotherrelation 2390
myquery 5
```

Reply:
This is too hard at the moment. The simplest way forward is the profiler.